test/unit/test_circuit_schedule.py

```diff
@@ -0,0 +1,199 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Unit tests for the circuit schedule class."""
+
+import ddt
+import numpy as np
+from qiskit_ibm_runtime.visualization.utils import plotly_module
+from qiskit_ibm_runtime.utils.circuit_schedule import CircuitSchedule
+from ..ibm_test_case import IBMTestCase
+from .mock.fake_circuit_schedule_timing import FakeCircuitScheduleInputData
+
+
+@ddt.ddt
+class TestCircuitSchedule(IBMTestCase):
+    """Tests for CircuitSchedule class."""
+
+    def setUp(self) -> None:
+        """Set up."""
+        fake_sampler_pub_result = FakeCircuitScheduleInputData.sampler_pub_result
+        self.circuit_schedule_data = fake_sampler_pub_result.metadata["compilation"][
+            "scheduler_timing"
+        ]["timing"].split("\n")
```

Comment:
nit - I should be able to pass the `timing` metadata directly to the drawer, without having to do `.split("\n")`

Reply:
I agree with Jessie, I think that the change could be as simple as allowing for the full string in `circuit_schedule`:

```
fig = draw_circuit_schedule_timing(
            circuit_schedule=circuit_schedule,
            included_channels=included_channels,
            filter_readout_channels=filter_readout_channels,
            filter_barriers=filter_barriers,
        )
```

and then splitting it inside `draw_circuit_schedule_timing`. This internal logic can be updated to accept changes in the string without changing much the interface.